### PR TITLE
feat: merge swipe and tap under the same button

### PR DESCRIPTION
**Pull request by alvarolaserna**
This PR is about the enhancement proposed in #1103

### CODE REVIEW

**Comment by eglitise** [comment]
There's quite a few things that I would like to be changed here, so instead I've made a branch of my own based on this code. [You can check the changes here](https://github.com/appium/appium-inspector/compare/main...eglitise:appium-inspector:merge-tap-and-swipe?expand=1). Please let me know if it looks okay from your side, behavior-wise.
Note that I haven't created a PR yet, since I still need to test this on iOS (and swiping is somehow broken in Firefox).

**Comment by alvarolaserna** [comment]
> There's quite a few things that I would like to be changed here, so instead I've made a branch of my own based on this code. [You can check the changes here](https://github.com/appium/appium-inspector/compare/main...eglitise:appium-inspector:merge-tap-and-swipe?expand=1). Please let me know if it looks okay from your side, behavior-wise. Note that I haven't created a PR yet, since I still need to test this on iOS (and swiping is somehow broken in Firefox).

I've made all the necessary changes so that it works as you intended, and also I have rename variables so that they make more sense. I have tested it on Firefox/Chrome and iOS/Android, both in mjpeg server mode and non-mjpeg.

File: app/renderer/components/Inspector/Screenshot.js [hunk 1]
```diff
@@ -1,70 +1,80 @@
...
-    } else if (screenshotInteractionMode === SWIPE) {
-      if (!swipeStart) {
-        setSwipeStart(x, y);
-      } else if (!swipeEnd) {
-        setSwipeEnd(x, y);
-        await B.delay(500); // Wait a second to do the swipe so user can see the SVG line
-        await handleDoSwipe({x, y}); // Pass swipeEnd because otherwise it is not retrieved
+  const handleScreenshotDown = async () => {
+    if (screenshotInteractionMode === TAP_SWIPE) {
+      await setCoordStart(x, y);
```
Comment (eglitise) [inline app/renderer/components/Inspector/Screenshot.js:32]: I would put `const { setCoordStart } = props;` at the start of this method, and remove it from the top-level `props` destructuring call.

File: app/renderer/components/Inspector/Screenshot.js [hunk 2]
```diff
@@ -1,70 +1,80 @@
...
+    }
+  };
+
+  const handleScreenshotUp = async () => {
+
```
Comment (eglitise) [inline app/renderer/components/Inspector/Screenshot.js:37]: Like the above, I would put `const { setCoordEnd } = props;` at the start of this method, and remove it from the top-level `props` destructuring call.

File: app/renderer/components/Inspector/Screenshot.js [hunk 3]
```diff
@@ -1,70 +1,80 @@
...
+    if (screenshotInteractionMode === TAP_SWIPE) {
+      await setCoordEnd(x, y);
+      if (Math.abs(coordStart.x - x) < 5 && Math.abs(coordStart.y - y) < 5) {
+        await handleDoTap({x, y}); // Pass coordEnd because otherwise it is not retrieved
+      } else {
+        await handleDoSwipe({x, y}); // Pass coordEnd because otherwise it is not retrieved
       }
     }
```
Comment (eglitise) [inline app/renderer/components/Inspector/Screenshot.js:45]: Would suggest putting `clearCoordAction();` here and removing it from `handleDoTap` and `handleDoSwipe`. You can then also retrieve `clearCoordAction` in a destructuring call at the start of this method, and remove it from the top-level `props` destructuring call.

**Review by eglitise — COMMENTED** [review]
This looks really good and works as expected! One minor thing is that on Firefox, the circle for swipe start point was either not drawn, or was the same size as the line, but this is not a big deal.
I will test this on iOS tomorrow and then approve :)

File: app/renderer/components/Inspector/Screenshot.js (hunk 3 above)
Comment (alvarolaserna) [reply app/renderer/components/Inspector/Screenshot.js:47]: Yes, agree on all the comments, fixed

File: app/renderer/components/Inspector/Screenshot.js (hunk 2 above)
Comment (alvarolaserna) [reply app/renderer/components/Inspector/Screenshot.js]: fixed

**Review by eglitise — APPROVED** [review]

**Comment by jlipps** [comment]
thanks a ton @alvarolaserna !

**Pull request merged**